File: 0874-backspace-string-compare/0874-backspace-string-compare.go

// processString processes the input string by simulating backspace operations
func processString(str string) string {
    // Initialize an empty slice to hold the processed characters
	var processed []rune

    for _, c := range str{
        if c == '#' {
            if len(processed) > 0 {
                processed = processed[:len(processed)-1]
            }
        } else {
            processed = append(processed,c)
        }
    }

    return string(processed)
}

func backspaceCompare(s string, t string) bool {
    // Process both strings and compare their results
	return processString(s) == processString(t)
}

File: 0874-backspace-string-compare/0874-backspace-string-compare.swift
class Solution {
    func processString(_ str: String) -> String {
        var processed = ""

        for c in str {
            if c == "#"{
                if !processed.isEmpty {
                    processed.removeLast()
                }
            } else {
                processed.append(c)
            }
        }

        return processed
    }

    func backspaceCompare(_ s: String, _ t: String) -> Bool {
        // Compare the processed versions of both strings
        return processString(s) == processString(t)
    }
}